src/compiler/moduleSpecifiers.ts

```diff
@@ -548,8 +548,10 @@ function getLocalModuleSpecifier(moduleFileName: string, info: Info, compilerOpt
             return maybeNonRelative;
         }
 
-        const nearestTargetPackageJson = getNearestAncestorDirectoryWithPackageJson(host, getDirectoryPath(modulePath));
-        const nearestSourcePackageJson = getNearestAncestorDirectoryWithPackageJson(host, sourceDirectory);
+        let nearestTargetPackageJson = getNearestAncestorDirectoryWithPackageJson(host, getDirectoryPath(modulePath));
+        let nearestSourcePackageJson = getNearestAncestorDirectoryWithPackageJson(host, sourceDirectory);
+        nearestTargetPackageJson &&= toPath(nearestTargetPackageJson, projectDirectory, getCanonicalFileName);
+        nearestSourcePackageJson &&= toPath(nearestSourcePackageJson, projectDirectory, getCanonicalFileName);

```

Comment:
Not clear if `getNearestAncestorDirectoryWithPackageJson` should just always return Path.

Reply:
Or if i should just `comparePaths`?

Reply:
This is used elsewhere and a `toPath` might have other negative affects at those use sites.

Reply:
If you mean my `Path` comment, it's only used when reading a file so shouldn't actually affect anything, but it sounds like the comparison one may just be better.

Reply:
I'd say just use `comparePaths`. We use it throughout `tryGetModuleNameFromExportsOrImports` against a path that comes from `getNearestAncestorDirectoryWithPackageJson` as well, so it would be consistent.

Reply:
Updated to use a helper; slightly different than the one you suggested but I was having trouble understanding what I was reading...